pkg/receive/multitsdb.go

```diff
@@ -493,6 +498,29 @@ func (t *MultiTSDB) startTSDB(logger log.Logger, tenantID string, tenant *tenant
 	reg = NewUnRegisterer(reg)
 
 	lset := labelpb.ExtendSortedLabels(t.labels, labels.FromStrings(t.tenantLabelName, tenantID))
+
+	hcLoop:
+	for hc := range t.hashringConfigs {
+		for t := range hc.Tenants {
```

Comment:
We need element here not index.
```suggestion
	for _, hc := range t.hashringConfigs {
		for _, t := range hc.Tenants {
```